nifi-nar-bundles/nifi-shopify-bundle/nifi-shopify-processors/src/main/java/org/apache/nifi/processors/shopify/GetShopify.java

```diff
@@ -0,0 +1,377 @@
+        final String limit = context.getProperty(RESULT_LIMIT).getValue();
+
+        shopifyResource = resourceType.getResource(resourceName);
+
+        shopifyRestService = getShopifyRestService(webClientService, uriBuilder, apiVersion, baseUrl, accessToken, resourceName,
+                limit, shopifyResource.getIncrementalLoadingParameter());
+    }
+
+    ShopifyRestService getShopifyRestService(final WebClientService webClientService, final HttpUriBuilder uriBuilder,
+                                             final String apiVersion, final String baseUrl, final String accessToken, final String resourceName,
+                                             final String limit, final IncrementalLoadingParameter incrementalLoadingParameter) {
+        return new ShopifyRestService(
+                webClientService,
+                uriBuilder,
+                apiVersion,
+                baseUrl,
+                accessToken,
+                resourceName,
+                limit,
+                incrementalLoadingParameter
+        );
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final StateMap state = getState(context);
+        final boolean isIncremental = context.getProperty(IS_INCREMENTAL).asBoolean();
+        final String initialStartTime = context.getProperty(INITIAL_INCREMENTAL_FILTER).getValue();
+        final Long incrDelayMs = context.getProperty(INCREMENTAL_DELAY).asTimePeriod(TimeUnit.MILLISECONDS);
+
+        String lastExecutionTime = state.get(resourceName);
+        if (lastExecutionTime == null && initialStartTime != null) {
+            lastExecutionTime = initialStartTime;
+        }
+
+        Instant now = getCurrentExecutionTime();
+        if (incrDelayMs != null) {
+            now = now.minus(incrDelayMs, ChronoUnit.MILLIS);
+        }
+        final String currentExecutionTime = now.toString();
+
+        String cursor = null;
+        do {
+            final AtomicInteger objectCountHolder = new AtomicInteger();
+            try (HttpResponseEntity response = shopifyRestService.getShopifyObjects(isIncremental, lastExecutionTime, currentExecutionTime, cursor)) {
+                cursor = getPageCursor(response);
+                if (response.statusCode() == HttpResponseStatus.OK.getCode()) {
+                    FlowFile flowFile = session.create();
+                    flowFile = session.write(flowFile, parseHttpResponse(response, objectCountHolder));
+                    if (objectCountHolder.get() > 0) {
+                        session.transfer(flowFile, REL_SUCCESS);
+                        if (cursor == null && isIncremental) {
```

Comment:
```suggestion
                        if (cursor == null && isIncremental && shopifyResource.isPaginationSupported()) {
```

Reply:
I think this check is at the wrong place. This branch means no pagination but to store the processing time in the state. I think we don't need the isPaginationSupported anymore since the while (cursor != null); will check if a cursor URL is returned from the response header.